.github/workflows/autopublish.yaml

```diff
@@ -0,0 +1,61 @@
+# To create a release, create a tag and push it to GitHub:
+#git tag -a "v0.0.1-beta" -m "beta version testing"
+#git push --tags
+# https://dev.to/iamtekson/publish-package-to-pypi-and-release-new-version-using-github-actions-108k
+name: Publish LitGPT to PyPI / GitHub
+
+on:
+  push:
+    tags:
+      - "v*"
+jobs:
+  build-n-publish:
+    name: Build and publish to PyPI
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
+
+      - name: Build source and wheel distributions
+        run: |
+          python -m pip install --upgrade build twine
+          python -m build
+          twine check --strict dist/*
+      - name: Publish distribution to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_API_TOKEN }}
+
+      - name: Create GitHub Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # This token is provided by Actions, you do not need to create your own token
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: ${{ github.ref }}
+          draft: false
+          prerelease: false
+
+      - name: Get Asset name
+        run: |
+          export PKG=$(ls dist/ | grep tar)
+          set -- $PKG
+          echo "name=$1" >> $GITHUB_ENV
+      - name: Upload Release Asset (sdist) to GitHub
+        id: upload-release-asset
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: dist/${{ env.name }}
+          asset_name: ${{ env.name }}
+          asset_content_type: application/zip
```

Comment:
Why should this workflow create a release automatically? Historically we've done this manually since we had to add release notes and such

Reply:
This is only triggered if you make a new tag, not each day or each PR merge or so. I thought this would be convenient.

Reply:
I see your motivation. I just don't know if we want to create releases like this automatically. Nowadays GitHub shows these on the timeline of users and such, so it might be a good idea not to do them unless we spend some time on the marketing and release notes etc

Reply:
Ok I see. My worry is that LitGPT is always moving relatively quickly and making it easy to release frequently would be nice. I can update this to not trigger automatically.

Reply:
I agree. Creating a tag and pushing to pypi is what the linked issue proposes. My comment above is only about not creating GitHub releases too.

Reply:
Ohhh, I see now what you mean. Got it now!

Reply:
I removed the publish to Gh part.